index.js

```diff
@@ -532,9 +558,17 @@
 
   //  Number$prototype$equals :: Number ~> Number -> Boolean
   function Number$prototype$equals(other) {
-    return typeof this === 'object' ?
-      equals(this.valueOf(), other.valueOf()) :
-      isNaN(this) && isNaN(other) || this === other && 1 / this === 1 / other;
+    return typeof other === 'object' ?
+      typeof this === 'object' &&
+        equals(this.valueOf(), other.valueOf()) :
+      isNaN(other) && isNaN(this) || this === other;
+  }
```

Comment:
The only change to this function should be the removal of the `1 / this === 1 / other` check.